### PR TITLE
Refactored disco_local_features hook to a gen_hook format

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all handlers for `disco_local_features` hook to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/136390) / small_tests / 3d55ec5fa40380d051297707f5174fe4ac08da5b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3758/136390/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3758/136390/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-13_11.06.18/lib.mongooseim.logs/run.2022-09-13_11.06.19/suite.log.html)

File: src/ejabberd_local.erl [hunk 1]
```diff
@@ -54,15 +54,15 @@
...
+-export([disco_local_features/3]).
 
 %% gen_server callbacks
 -export([init/1, handle_call/3, handle_cast/2, handle_info/2,
          terminate/2, code_change/3]).
 
 -export([do_route/4]).
 
--ignore_xref([disco_local_features/1, do_route/4, get_iq_callback/1,
+-ignore_xref([disco_local_features/3, do_route/4, get_iq_callback/1,
```
Comment (NelsonVides) [inline src/ejabberd_local.erl:65]: Actually, xref can see when you do `fun ?MODULE:disco_local_features/3` that the function is actually used, maybe not called, but given elsewhere who might do something with it, so it's not dead code. So you can actually remove all there `disco_local_features/3` from the xref ignores everywhere 😄

**Review by NelsonVides — COMMENTED** [review]
I like the renaming of all hooks to `legacy_hooks`, make it clear who's leaving here 😛 
Only one comment on xref and then this is ready, great job! 👌🏽

**Comment by DenysGonchar** [comment]
Thanks for the PR, it looks fine.

I have one note for the future changes, taking into account that some of the hooks are quite generic and used in a lot of modules, you may want to have simultaneous support of the new `gen_hook` and the legacy `ejabberd_hooks` interfaces for a short period of time.  

e.g. the `disco_local_features` hook execution function could look like this: 
```
-spec disco_local_features(mongoose_disco:feature_acc()) -> mongoose_disco:feature_acc().
disco_local_features(Acc = #{host_type := HostType}) ->
    Parameters = #{},
    LegacyArgs = [Node],
    ParametersWithLegacyArgs = ejabberd_hooks:add_args(Params, LegacyArgs),
    run_hook_for_host_type(disco_local_features, HostType, Acc, ParametersWithLegacyArgs).
```

And it would work for both types (new and legacy) of interfaces, And it would work for both types (new and legacy) of interfaces, allowing changing the corresponding modules in the multiple PRs. And so, organize rework on a per-module basis rather than per-hook.

Once all the modules are converted, we can make a finalizing PR which removes `ejabberd_hooks` module and `ejabberd_hooks:add_args/2` calls from all the hook execution wrappers in the `mongoose_hooks` modules.

**Comment by mongoose-im** [comment]
[riak_mnesia_24](https://circleci.com/gh/esl/MongooseIM/136601) / riak_mnesia / ecd29ce3a2aebade6f98a189f16c284923b4280f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3758/136601/riak_mnesia.24.3.4)

**Review by DenysGonchar — APPROVED** [review]
Looks fine to me

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

**Review by chrzaszcz — APPROVED** [review]
Looks good :+1:

**Pull request merged**